### DISCUSSION TITLE
How can I switch pages?

### DISCUSSION THREAD

**Discussion by Tryanks** [AvaloniaUI/Avalonia#9881] [Q&A] [answered]
Hi,

I have a requirement that when I start the software, the start page is displayed and has a button to select a file. After clicking the button to select a file, then the start page closes and goes to the page used to process the file.

The whole process is done in one window without opening a new one.

I am still just a newbie to Avalonia and I didn't find a control similar to WPF's `Frame` in Avalonia, I would like to know what is a similar solution on the Avalonia side? Are there some examples to learn from?

**Comment by timunie** [comment, marked as answer]
I'd use different `UserControls` as pages. You can bind the content to a `PageViewModel` and select the `DataTemplate` besed on the current page. See https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/Routing/BasicViewLocatorSample for reference.
